transformer_engine/common/util/pybind_helper.h

```diff
@@ -0,0 +1,106 @@
+/*************************************************************************
+ * Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See LICENSE for license information.
+ ************************************************************************/
+
+#ifndef TRANSFORMER_ENGINE_COMMON_PYBIND_HELPER_H_
+#define TRANSFORMER_ENGINE_COMMON_PYBIND_HELPER_H_
+
+#include <pybind11/functional.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <transformer_engine/activation.h>
+#include <transformer_engine/comm_gemm_overlap.h>
+#include <transformer_engine/fused_attn.h>
+#include <transformer_engine/transformer_engine.h>
+
+// NOTE: transformer_engine::DType is a strongly-typed enum class so it should not have
+//       `.export_values()`. NVTE_ prefixed enums are classic C enums to make core headers
+//       includable in external pure-C libraries, and they need `.export_values()` to work
+//       correctly in Python.
```

Comment:
As far as I can tell from the Pybind11 [docs](https://pybind11.readthedocs.io/en/stable/classes.html) and [tests](https://github.com/pybind/pybind11/blob/723307283ed7dc03c901fba1da1127c7b16a4a0d/tests/test_enum.py#L10-L15), `.export_values()` doesn't affect anything C++-related (e.g. enum vs enum class), and it only affects whether the Python enums are unscoped. However, we already use scoped enums, e.g.:
https://github.com/NVIDIA/TransformerEngine/blob/91a16a3f9c8c1aedd4cd0b1da7a0da58e977412c/transformer_engine/jax/cpp_extensions/activation.py#L33-L44
`.export_values()` allows us to do:
```python
 ActivationEnum = { 
     ("gelu",): transformer_engine_jax.GELU, 
     ("gelu", "linear"): transformer_engine_jax.GEGLU, 
```
I don't think this is a good direction though. Scoped enums are more explicit, and the only reason we have unscoped enums in C++ is because of C compatibility.

Reply:
I think this was leftover from an earlier iteration of the refactor that compiled a `transformer_engine_common.so` PyBind11 extension, which was then imported into `transformer_engine_pytorch.so` and other framework extensions. In that particular use case, omitting `export_values()` caused TE/common enums to register as different data types than what was expected by TE/PyTorch functions even though they're the same C enum underneath.

That's no longer relevant because the refactor no longer compiles a `transformer_engine_common` extension. Instead, we just have a helper macro that adds TE/common definitions into the PyBind11 definitions for framework extensions. As far as the framework extension is concerned, the enums are native and recognized correctly without `export_values()`. So I'll go ahead and remove these now.